TELCODOCS-778 OVN/OVS Troubleshooting guide

kquinn1204 opened the pull request:
[TELCODOCS-778]:OVN/OVS Troubleshooting guide

Version(s):Main, 4.12 and 4.13
Issue:
https://issues.redhat.com/browse/TELCODOCS-778

Link to docs preview:

- https://54155--docspreview.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.html 
- https://54155--docspreview.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/ovn-kubernetes-architecture-assembly.html
- https://54155--docspreview.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/ovn-kubernetes-troubleshooting-sources.html
- https://54155--docspreview.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/ovn-kubernetes-tracing-using-ovntrace.html

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

CFields651 commented:
About section:
- on this note: "OVN-Kubernetes is the default networking solution for OpenShift Container Platform and single-node OpenShift deployments;" is it okay to say that's new with 4.12?  

OVN-K arch section: (key components bullet points)
- The "OVN-Northd" component runs in the 'northd' container in the ovnkube-master pods
- The "OVN-Controller" component runs in the 'ovn-controller' container of the ovnkube-node pods. The "ovn-controller" container is mentioned later in the text.  
- This sentence is a bit awkward: "The OVN northbound database stores objects for example logical routers," -maybe a comma after "routers"? 
- This sentenced is a bit confusing.  I think it could be eliminated because what is intended is already said in the paragraph: " The first daemon set matches on the node role master, so that matches on three masters, and the next one matches on all of them."

CFields651 commented:
Overall, this is just what we talked about in https://issues.redhat.com/browse/TELCODOCS-778 so I like the layout and the content.    It is an appropriate subset of the end-to-end troubleshooting guide.

CFields651 commented on modules/nw-ovn-kubernetes-get-raft-db-status.adoc:
```diff
@@ -0,0 +1,121 @@
++
+[source,terminal]
+----
+$ oc get all -n openshift-ovn-kubernetes
```
Already showed how to get all the resources in the "Listing all resources in the OVN-Kubernetes project" above.  I feel this command is not necessary.

kquinn1204 replied:
@CFields651 I guess I feel maybe user will not read doc in linear fashion and might just see this section in TOC so they will simply go to this section, so I was keeping it all self-contained. Personally if I go to a section I like for the most part having everything I need there within reason

CFields651 commented on modules/nw-ovn-kubernetes-get-raft-db-status.adoc:
```diff
@@ -0,0 +1,121 @@
+daemonset.apps/ovnkube-node     6         6         6       6            6           beta.kubernetes.io/os=linux                                   133m
+----
+
+. Select one of the `ovnkube-master` pods and run the following to determine the status of the OVN northbound database:
```
We're really getting the role on a node.  One of them must be the 'leader.'  If not then we have an issue.  You might reference that there is a readinessProbe for this and link to that section.

kquinn1204 replied:
updated

CFields651 commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-architecture.adoc:
```diff
@@ -0,0 +1,51 @@
+* **Cloud Management System (CMS)** - A platform specific client for OVN that provides a CMS specific plugin for OVN integration. The plugin translates the cloud management system's concept of the logical network configuration, stored in the CMS configuration database in a  CMS-specific  format, into an intermediate representation understood by OVN.
+* **OVN Northbound database (`nbdb`)** - Stores the logical network configuration passed by the CMS plugin.
+* **OVN Southbound database (`sbdb`)** - Stores the physical and logical network configuration state for OpenVswitch (OVS) system on each node, including tables that bind them.
+* **ovn-northd** - This is the intermediary client between `nbdb` and `sbdb`. It translates  the logical network configuration in terms of conventional network concepts, taken from the `nbdb`, into  logical data path flows in the `sbdb` below it. The `ovn-northd` component runs in the `northd` container in the `ovnkube-master` pods.
```
container name is 'northd'

CFields651 commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-architecture.adoc:
```diff
@@ -0,0 +1,51 @@
+* **OVN Northbound database (`nbdb`)** - Stores the logical network configuration passed by the CMS plugin.
+* **OVN Southbound database (`sbdb`)** - Stores the physical and logical network configuration state for OpenVswitch (OVS) system on each node, including tables that bind them.
+* **ovn-northd** - This is the intermediary client between `nbdb` and `sbdb`. It translates  the logical network configuration in terms of conventional network concepts, taken from the `nbdb`, into  logical data path flows in the `sbdb` below it. The `ovn-northd` component runs in the `northd` container in the `ovnkube-master` pods.
+* **ovn-controller** - This is the OVN agent that interacts with OVS and hypervisors, for any information or update that is needed for `sbdb`. The `ovn-controller` reads logical flows from the `sbdb`, translates them into `OpenFlow` flows and sends them to the node’s OVS daemon. The `ovn-controller` component runs in the `ovn-controller` container of the `ovnkube-node` pods.
```
container name is the same 'ovn-controller'

CFields651 commented on modules/nw-ovn-kubernetes-list-database-contents.adoc:
```diff
@@ -0,0 +1,218 @@
+
+.Procedure
+
+. Find the OVN raft leader for the northbound database.
```
We already showed how to find the raft leader in the "Querying the status of the northbound and southbound databases" section.  I don't think we need to repeat it here.

kquinn1204 replied:
@CFields651 I guess I feel maybe user will not read doc in linear fashion and might just see this section in TOC so they will simply go to this section, so I was keeping it all self-contained. Personally if I go to a section I like for the most part having everything I need there within reason

kquinn1204 replied:
@CFields651 I want to keep every section self-contained, I assume customers will not read linearly

CFields651 commented on modules/nw-ovn-kubernetes-list-southbound-database-contents.adoc:
```diff
@@ -0,0 +1,141 @@
+
+.Procedure
+
+. Find the OVN raft leader for the southbound database.
```
Already showed how to do this in the "Querying the status of the northbound and southbound databases" section.  Do we really want to repeat it?

kquinn1204 replied:
@CFields651 I think I do, as I like sections to stand on their own as much as possible. I guess I feel customers will not read this linearly, and I added to the previous section more as an intro to the table of reference command line arguments. Previous section does not have the full details

kquinn1204 replied:
@CFields651 I guess I feel like each section should stand on its own for the most part, as this is a procedure someone may just go to from the TOC without reading the previous section.

CFields651 commented on modules/nw-ovn-kubernetes-readiness-probes.adoc:
```diff
@@ -0,0 +1,60 @@
+[id="nw-ovn-kubernetes-project-readiness-probes_{context}"]
+= OVN-Kubernetes readiness probes
+
+The `ovnkube-master` and `ovnkube-node` pods have containers configured with readiness probes.
```
would be nice to link to the section of the docs that describe readinessProbes at https://docs.openshift.com/container-platform/4.12/applications/application-health.html.

kquinn1204 replied:
link added

CFields651 commented on modules/nw-ovn-kubernetes-pod-connectivity-checks.adoc:
```diff
@@ -0,0 +1,64 @@
+[id="nw-ovn-kubernetes-pod-network-connectivity-checks_{context}"]
+= OVN-Kubernetes pod network connectivity checks
+
+The connectivity check controller, in {product-title} 4.10 and later, orchestrates connection verification checks in your cluster. These include Kubernetes API, OpenShift API and individual nodes. The results for the connection tests are stored in `PodNetworkConnectivity` objects in the `openshift-network-diagnostics` namespace. Connection tests are performed every minute in parallel.
```
would be good to link to where this is described in the Networking guide at https://access.redhat.com/documentation/en-us/openshift_container_platform/4.12/html/networking/verifying-connectivity-endpoint

kquinn1204 commented:
/label telco

kquinn1204 commented:
/label peer-review-needed

michaelryanpeter commented:
/label peer-review-in-progress

michaelryanpeter commented on modules/nw-ovn-kubernetes-alerts.adoc:
```diff
@@ -0,0 +1,52 @@
+
+. . View active or firing alerts by running the following commands:
++
+[source,terminal]
+----
+ALERT_MANAGER=$(oc get route alertmanager-main -n openshift-monitoring -o jsonpath='{@.spec.host}')
+----
++
+[source,terminal]
+----
+curl -s -k -H "Authorization: Bearer $(oc create token prometheus-k8s -n openshift-monitoring)" https://$ALERT_MANAGER/api/v1/alerts | jq '.data[] | "\(.labels.severity) \(.labels.alertname) \(.labels.pod) \(.labels.container) \(.labels.endpoint) \(.labels.instance)"'
+----
```
See the [repo guidance ](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output)on breaking up long lines of code.

skrthomas commented on networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.adoc:
```diff
@@ -18,6 +18,21 @@ OVN configures OVS on each node to implement the declared network configuration.
 OVN-Kubernetes is the default networking solution for {product-title} and {sno} deployments.
 ====
 
+OVN-Kubernetes, which arose from the OVS project, uses many of the same constructs, such as open flow rules, to determine how packets will travel through the network.
```
```suggestion
OVN-Kubernetes, which arose from the OVS project, uses many of the same constructs, such as open flow rules, to determine how packets travel through the network.
```

skrthomas replied:
future tense

skrthomas commented on networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.adoc:
```diff
@@ -18,6 +18,21 @@ OVN configures OVS on each node to implement the declared network configuration.
+So for example if you have a pod that sends out a DHCP request on the network, it sends out that broadcast looking for DHCP address there will be a logic flow rule that matches that packet, and it responds giving it a gateway, a DNS server an IP address and so on.
+
+OVN-Kubernetes runs a daemon on each node. There are daemon sets for the databases and for the OVN controller that run on every node.
+The OVN controller programs the OVS switch on the nodes to support the network provider features; egress IPs, firewalls, routers, hybrid networking, IPSEC encryption, IPv6, network policy, network policy logs, hardware offloading and multicast.
```
```suggestion
The OVN controller programs the OVS turn on the nodes to support the network provider features; egress IPs, firewalls, routers, hybrid networking, IPSEC encryption, IPv6, network policy, network policy logs, hardware offloading and multicast.
```

skrthomas replied:
Vale is flagging switch on as a Red Hat suggested language to use turn on instead.

skrthomas commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-logical-architecture.adoc:
```diff
@@ -0,0 +1,29 @@
+The key components are:
+
+* **Gateway Routers** - (sometimes called L3 Gateway Routers) are typically used between distributed routers and the physical network. It (including its logical patch ports) is bound to a physical location (not distributed), or chassis. The patch ports on this router are known as l3gateway ports in ovn-sbdb. example: GR_ovn-worker1 and GR_ovn-worker2.
+* **Distributed Logical Routers** - and the logical switches behind it, to which VMs and containers attach, effectively reside on each hypervisor. example: ovn_cluster_router.
```
```suggestion
* **Distributed Logical Routers** - and the logical switches behind it, to which virtual machines and containers attach, effectively reside on each hypervisor. example: ovn_cluster_router.
```

kquinn1204 replied:
removed file as copied contents to modules

skrthomas commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-logical-architecture.adoc:
```diff
@@ -0,0 +1,29 @@
+* **localnet ports** - are present on the bridged logical switches that allows a connection to a locally accessible network from each ovn-controller instance. This helps model direct connectivity to the physical network from the logical switches. A logical switch can only have a single localnet port attached to it.
+
+
+[role="_additional-resources"]
```
Is this meant to be empty?

kquinn1204 replied:
removed file as copied contents to modules

skrthomas commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-logical-architecture.adoc:
```diff
@@ -0,0 +1,29 @@
+* **Gateway Routers** - (sometimes called L3 Gateway Routers) are typically used between distributed routers and the physical network. It (including its logical patch ports) is bound to a physical location (not distributed), or chassis. The patch ports on this router are known as l3gateway ports in ovn-sbdb. example: GR_ovn-worker1 and GR_ovn-worker2.
+* **Distributed Logical Routers** - and the logical switches behind it, to which VMs and containers attach, effectively reside on each hypervisor. example: ovn_cluster_router.
+* **Join local switch** - used to connect the distributed router and gateway routers. It reduces the number of IP addresses needed on the distributed router. example: join.
+* **Logical switches with patch ports** - are used to virtualize the network stack. They connect remote logical ports through tunnels. example: ovn-worker && ovn-worker2.
```
Virtualize is not a word....consider how else to describe this

skrthomas commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-logical-architecture.adoc:
```diff
@@ -0,0 +1,29 @@
+* **Distributed Logical Routers** - and the logical switches behind it, to which VMs and containers attach, effectively reside on each hypervisor. example: ovn_cluster_router.
+* **Join local switch** - used to connect the distributed router and gateway routers. It reduces the number of IP addresses needed on the distributed router. example: join.
+* **Logical switches with patch ports** - are used to virtualize the network stack. They connect remote logical ports through tunnels. example: ovn-worker && ovn-worker2.
+* **Logical switches with localnet ports** - are used to bridge/connect OVN to the physical network. They connect remote logical ports by bridging the packets to directly connected physical L2 segments using localnet ports. example: ext_ovn-worker1.
```
```suggestion
* **Logical switches with localnet ports** - are used to bridge and connect OVN to the physical network. They connect remote logical ports by bridging the packets to directly connected physical L2 segments using localnet ports. example: ext_ovn-worker1.
```

skrthomas replied:
use and or or instead of /

kquinn1204 replied:
removed file as copied contents to modules

michaelryanpeter commented on modules/nw-ovn-kubernetes-examine-database-contents-ref.adoc:
```diff
@@ -0,0 +1,73 @@
+$ oc exec -n openshift-ovn-kubernetes -it ovnkube-master-mk6p6 -c northd -- ovn-sbctl show
+----
+
+Additional command line arguments are described in this table:
```
We are supposed to avoid self-referential language. See the [ISG guidance](https://www.ibm.com/docs/en/ibm-style?topic=format-tables#references) on table references for suggestions on table introductions.

skrthomas commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-logical-architecture.adoc:
```diff
@@ -0,0 +1,29 @@
+* **Logical switches with patch ports** - are used to virtualize the network stack. They connect remote logical ports through tunnels. example: ovn-worker && ovn-worker2.
+* **Logical switches with localnet ports** - are used to bridge/connect OVN to the physical network. They connect remote logical ports by bridging the packets to directly connected physical L2 segments using localnet ports. example: ext_ovn-worker1.
+* **Patch ports** - represent connectivity between logical switches and logical routers and between peer logical routers. A single connection will have a pair of patch ports at each such point of connectivity, one on each side.
+* **l3gateway ports** - are the port binding entries in sddb for logical patch ports used in the gateway routers. They are called  l3gateway ports rather than patch ports just to portray the fact that these ports are bound to a chassis just like the gateway router itself.
```
what does sddb stand for? May want to spell out the acronym since it isn't used on this page.

kquinn1204 replied:
removed file as copied contents to modules

skrthomas reviewed (commented):
A few comments

michaelryanpeter reviewed (commented):
I completed a review of the following modules:
- nw-ovn-kubernetes-alerts.adoc: https://github.com/openshift/openshift-docs/pull/54155/files#diff-78f7fa426f00a975812cd692d1cc691c3404c84d05d9bdf7b9117380d279fd85
- nw-ovn-kubernetes-change-log-levels.adoc: https://github.com/openshift/openshift-docs/pull/54155/files#diff-2b1dbe2d3109236232b464783bfa0cf20e7e81e8e20c8a1d05f096c54bdb9987
- nw-ovn-kuberentes-examine-database-contents-ref.adoc: https://github.com/openshift/openshift-docs/pull/54155/files#diff-f727dfec8e363f15050ebbf41328ba45ea0925fdea85bc6e087156744eabccba

I understand that this is not always possible, but, in the future, please consider breaking up PRs of this scope into smaller PRs. While we are trying to come up with processes to share reviews and "swarm" very large PRs, we are not there yet.

Please remember to squash your commits. I am removing the peer-review-in-progress label. Since the review is not complete, I am not putting adding the peer-review-done label. Please let me know if you have questions regarding the modules I reviewed.

/remove-label peer-review-in-progress

michaelryanpeter commented on modules/nw-ovn-kubernetes-examine-database-contents-ref.adoc:
```diff
@@ -0,0 +1,73 @@
+| `ovn-nbctl show`
+|An overview of the northbound database contents.
+
+| `ovn-nbctl show [switch \| router]`
```
I have not seen this format with brackets and a pipe as a variable in a table heading. I think that this should be our standard user supplied variable of `<switch_or_router>`. The same comment applies throughout the table where brackets are used.

michaelryanpeter commented on modules/nw-ovn-kubernetes-examine-database-contents-ref.adoc:
```diff
@@ -0,0 +1,73 @@
+| Show logical routers.
+
+| `ovn-nbctl lrp-list [router]`
+|Using the router information from the previous command drill down and show the router ports.
```
```suggestion
|Using the router information from the previous command, drill down and show the router ports.
```

kquinn1204 replied:
Reorganised most of this section into two distinct reference topics. The concept type information was in fact covered elsewhere.

maxwelldb commented on modules/nw-ovn-kubernetes-get-raft-db-status.adoc:
```diff
@@ -0,0 +1,125 @@
+// * networking/ovn_kubernetes_network_provider/ovn-kubernetes-architecture.adoc
+
+:_content-type: PROCEDURE
+[id="nw-list-querying-status-northbound-southbound-databases_{context}"]
```
This should match the file name.

kquinn1204 replied:
File removed as not in use

maxwelldb commented on modules/nw-ovn-kubernetes-get-raft-db-status.adoc:
```diff
@@ -0,0 +1,125 @@
+[id="nw-list-querying-status-northbound-southbound-databases_{context}"]
+= Querying the status of the northbound and southbound databases
+
+The Kubernetes control plane nodes each contain an `ovnkube-master` pod which hosts containers for the OVN northbound and southbound databases which must form `Raft` consensus between its peer database.
```
ISG: Use "that" rather than "which" in this case.

maxwelldb replied:
I'm having trouble parsing this sentence. Is the consensus formed between/with/by a single peer database?

maxwelldb replied:
This abstract doesn't seem like it's from a procedure to me. From the mod docs documentation:

> The procedure introduction is a short paragraph that provides an overview of the module. The introduction includes what the module will help the user do and why it will be beneficial to the user. It includes key words that relate to the module to maximize search engine optimization.

maxwelldb commented on modules/nw-ovn-kubernetes-get-raft-db-status.adoc:
```diff
@@ -0,0 +1,125 @@
+= Querying the status of the northbound and southbound databases
+
+The Kubernetes control plane nodes each contain an `ovnkube-master` pod which hosts containers for the OVN northbound and southbound databases which must form `Raft` consensus between its peer database.
+For example all OVN northbound databases must form a `Raft` cluster and all southbound databases must form a separate `Raft` cluster. At any given time a single `ovnkube-master` is the "leader" and
```
ISG: Avoid this sort of quotation mark use unless there is no alternative.

maxwelldb commented on modules/nw-ovn-kubernetes-get-raft-db-status.adoc:
```diff
@@ -0,0 +1,125 @@
+
+.Prerequisites
+
+* Access to the cluster as a user with the cluster-admin role.
```
```suggestion
* Access to the cluster as a user with the `cluster-admin` role.
```

maxwelldb commented on modules/nw-ovn-kubernetes-get-raft-db-status.adoc:
```diff
@@ -0,0 +1,125 @@
++
+<1> This pod is the leader.
+
+. Select one of the `ovnkube-master` pods and run the following to determine the role of that pod on a node::
```
```suggestion
. Select one of the `ovnkube-master` pods and run the following to determine the role of that pod on a node:
```

maxwelldb commented on modules/nw-ovn-kubernetes-install-ovnkube-trace-local.adoc:
```diff
@@ -0,0 +1,86 @@
+[id="nw-installing-ovnkube-localhost_{context}"]
+= Installing the ovnkube-trace on local host
+
+Follow this procedure to install `ovnkube-trace` on your local host.
```
This introduction is self-referential, and it's not in line with the mod docs guidelines.

https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs

maxwelldb commented on modules/nw-ovn-kubernetes-install-ovnkube-trace-local.adoc:
```diff
@@ -0,0 +1,86 @@
+    	use udp transport protocol
+----
++
+The command-line arguments supported are familiar Kubernetes constructs, such as namespaces, pods, services so you do not need to determine the MAC address, the IP address of the destination nodes or the ICMP type.
```
```suggestion
The command-line arguments supported are familiar Kubernetes constructs, such as namespaces, pods, services so you do not need to determine the MAC address, the IP address of the destination nodes, or the ICMP type.
```

maxwelldb commented on modules/nw-ovn-kubernetes-list-database-contents.adoc:
```diff
@@ -0,0 +1,227 @@
+
+.Prerequisites
+
+* Access to the cluster as a user with the cluster-admin role.
+* The OpenShift CLI (oc) installed.
+
+To understand logic flow rules you need to examine the northbound database and understand what objects are there to see how they are translated into logic flow rules.
+
+Follow this procedure to list OVN database contents with OVN-Kubernetes.
```
This is not formatted according to prereq guidelines. Rewrite into an UL.

kquinn1204 replied:
i redesigned this section

maxwelldb commented on modules/nw-ovn-kubernetes-list-resources.adoc:
```diff
@@ -0,0 +1,97 @@
+
+.Prerequisites
+
+* Access to the cluster as a user with the cluster-admin role.
+* The OpenShift CLI (oc) installed.
+
+.Procedure
+
+. Run the following command to get all resources, endpoints, and configmaps in the OVN-Kubernetes project:
```
```suggestion
* Access to the cluster as a user with the cluster-admin role.
* The OpenShift CLI (oc) installed.

.Procedure

. Run the following command to get all resources, endpoints, and config maps in the OVN-Kubernetes project:
```

kquinn1204 replied:
seems to be ConfigMaps in docs in other places

maxwelldb commented on modules/nw-ovn-kubernetes-list-resources.adoc:
```diff
@@ -0,0 +1,97 @@
+
+:_content-type: PROCEDURE
+[id="nw-list-all-resources-ovn-kubernetes-project_{context}"]
+= Listing all resources in the OVN-Kubernetes project
+
+.Prerequisites
+
```
There is no introduction here.

kquinn1204 replied:
intro added

maxwelldb commented on modules/nw-ovn-kubernetes-list-southbound-database-contents.adoc:
```diff
@@ -0,0 +1,150 @@
+
+:_content-type: PROCEDURE
+[id="nw-list-ovn-southbound-database-contents_{context}"]
+= Listing the OVN southbound database contents with ovn-kubernetes
+
+.Prerequisites
```
No intro?

kquinn1204 replied:
intro added

maxwelldb commented on modules/nw-ovn-kubernetes-list-southbound-database-contents.adoc:
```diff
@@ -0,0 +1,150 @@
+// * networking/ovn_kubernetes_network_provider/ovn-kubernetes-architecture.adoc
+
+:_content-type: PROCEDURE
+[id="nw-list-ovn-southbound-database-contents_{context}"]
```
Module anchors must always match file name.

maxwelldb commented on modules/nw-ovn-kubernetes-list-southbound-database-contents.adoc:
```diff
@@ -0,0 +1,150 @@
+.Prerequisites
+
+* Access to the cluster as a user with the cluster-admin role.
+* The OpenShift CLI (oc) installed.
```
```suggestion
* The OpenShift CLI (`oc`) installed.
```

maxwelldb commented on modules/nw-ovn-kubernetes-list-southbound-database-contents.adoc:
```diff
@@ -0,0 +1,150 @@
+* Access to the cluster as a user with the cluster-admin role.
+* The OpenShift CLI (oc) installed.
+
+Logic flow rules are stored in the southbound database which is a representation of your infrastructure. Every node in the cluster is a chassis and to view the  southbound database information use `ovn-sbctl show`.
+
+Follow this procedure to list OVN southbound database contents with OVN-Kubernetes.
```
Is this the intro? It should be above the prereqs.

maxwelldb commented on modules/nw-ovn-kubernetes-logs.adoc:
```diff
@@ -0,0 +1,76 @@
+
+* Access to the OpenShift CLI (`oc`).
+
+.Procedure (UI)
```
These are really two different modules--one for performing this procedure with a web UI, and the other with a CLI.

kquinn1204 replied:
Done

maxwelldb commented on modules/nw-ovn-kubernetes-logs.adoc:
```diff
@@ -0,0 +1,76 @@
+
+:_content-type: PROCEDURE
+[id="nw-ovn-kubernetes-logs_{context}"]
+= OVN-Kubernetes logs
```
This title should use the procedure module format from the mod docs guidelines.

kquinn1204 replied:
Rewritten to reflect this is a procedure

maxwelldb commented on modules/nw-ovn-kubernetes-readiness-probes.adoc:
```diff
@@ -0,0 +1,60 @@
+
+:_content-type: PROCEDURE
+[id="nw-ovn-kubernetes-project-readiness-probes_{context}"]
+= OVN-Kubernetes readiness probes
```
This title looks like it's for a concept module.

kquinn1204 replied:
Rewritten correctly to reflect that this is a procedure my bad

maxwelldb commented on modules/nw-ovn-kubernetes-resources-con.adoc:
```diff
@@ -0,0 +1,40 @@
+//
+// * networking/ovn_kubernetes_network_provider/ovn-kubernetes-architecture.adoc
+
+:_content-type: CONCEPT
+[id="nw-resources-ovn-kubernetes-project_{context}"]
+= Resources in the OVN-Kubernetes project
+
+The OVN-Kubernetes Container Network Interface (CNI) cluster network provider implements the following features:
+
+. Run the following command to get all resources in the OVN-Kubernetes project
++
+[source,terminal]
+----
+$ oc get all -n openshift-ovn-kubernetes
+----
```
Concept modules don't have steps in them.

kquinn1204 replied:
I removed this section as covered elsewhere

maxwelldb commented on modules/ovn-kubernetes-logical-architecture-con.adoc:
```diff
@@ -0,0 +1,20 @@
+
+The key components are:
+
+* **Gateway routers** - Gateway routers sometimes called L3 gateway routers, are typically used between distributed routers and the physical network. Gateway routers including their logical patch ports is bound to a physical location (not distributed), or chassis. The patch ports on this router are known as l3gateway ports in `ovn-sbdb`. For example `GR_ovn-worker1` and `GR_ovn-worker2`.
+* **Distributed logical routers** - Distributed logical Routers and the logical switches behind them, to which virtual machines and containers attach, effectively reside on each hypervisor. For example `ovn_cluster_router`.
+* **Join local switch** - Join local switches are used to connect the distributed router and gateway routers. It reduces the number of IP addresses needed on the distributed router. For example `join`.
+* **Logical switches with patch ports** - Logical switches with patch ports are used to virtualize the network stack. They connect remote logical ports through tunnels. For example `ovn-worker` and `ovn-worker2`.
+* **Logical switches with localnet ports** - Logical switches with localnet ports are used to connect OVN to the physical network. They connect remote logical ports by bridging the packets to directly connected physical L2 segments using localnet ports. For example `ext_ovn-worker1`.
+* **Patch ports** - Patch ports represent connectivity between logical switches and logical routers and between peer logical routers. A single connection has a pair of patch ports at each such point of connectivity, one on each side.
+* **l3gateway ports** - l3gateway ports are the port binding entries in `ovn-sbdb` for logical patch ports used in the gateway routers. They are called l3gateway ports rather than patch ports just to portray the fact that these ports are bound to a chassis just like the gateway router itself.
+* **localnet ports** - localnet ports are present on the bridged logical switches that allows a connection to a locally accessible network from each `ovn-controller` instance. This helps model the direct connectivity to the physical network from the logical switches. A logical switch can only have a single localnet port attached to it.
```
This is a very good place to use a description list: https://docs.asciidoctor.org/asciidoc/latest/lists/description/

kquinn1204 replied:
done

maxwelldb commented on modules/ovn-kubernetes-logical-architecture-con.adoc:
```diff
@@ -0,0 +1,20 @@
+The following diagram shows the OVN-Kubernetes logical architecture.
+
+.OVN-Kubernetes architecture
+image::299_OpenShift_OVN-Kubernetes_arch_0123_2.png[OVN-Kubernetes logical architecture]
```
This alt text doesn't really describe the image.

Here are some outside examples that might be helpful:

- https://www.w3.org/WAI/tutorials/images/complex/
- https://sc.edu/about/offices_and_divisions/digital-accessibility/guides_tutorials/alternative_text/charts-diagrams/index.php ("Example of Describing a Diagram")

Ignore the longdesc parts in the w3 site--there's no great way to do it with our toolchain, anyway, but "Bar chart showing blah blah blah" is fair.

maxwelldb commented on modules/nw-ovn-kubernetes-running-ovnkube-trace.adoc:
```diff
@@ -0,0 +1,306 @@
+<5> Use the `udp` transport protocol. Port 53 is the port the DNS service uses.
+<6> Set the log level to 1 (0 is minimal and 5 is debug)
++
+.Expected output
+
+[source,terminal]
```
This label won't apply to the code block due to the line break.

```suggestion
.Expected output
[source,terminal]
```

maxwelldb commented on modules/nw-ovn-kubernetes-list-southbound-database-contents.adoc:
```diff
@@ -0,0 +1,150 @@
+----
+$ oc exec -n openshift-ovn-kubernetes -it ovnkube-master-mk6p6 -c northd -- ovn-sbctl --help
+----
+
+
+
```
```suggestion

```

maxwelldb commented on modules/nw-ovn-kubernetes-pod-connectivity-checks.adoc:
```diff
@@ -0,0 +1,64 @@
+
+.Procedure
+
+. To list the current `PodNetworkConnectivityCheck` objects, enter the following command::
```
```suggestion
. To list the current `PodNetworkConnectivityCheck` objects, enter the following command:
```

maxwelldb commented on modules/nw-ovn-kubernetes-pod-connectivity-checks.adoc:
```diff
@@ -0,0 +1,64 @@
+
+:_content-type: PROCEDURE
+[id="nw-ovn-kubernetes-pod-network-connectivity-checks_{context}"]
+= OVN-Kubernetes pod network connectivity checks
```
This procedure looks like it's full of a lot of one-item procedure modules to me.

kquinn1204 replied:
@maxwelldb I think it is valid as they are all connectivity checks, I do think it would not look great if split into a lot of one-item procedure modules.

maxwelldb commented on modules/ovn-kubernetes-logical-architecture-con.adoc:
```diff
@@ -0,0 +1,20 @@
+[id="ovn-kubernetes-logical-architecture-con_{context}"]
+= OVN-Kubernetes logical architecture
+
+The following diagram shows the OVN-Kubernetes logical architecture.
```
Without a more comprehensive description of the architecture that this module is about, it's not accessible. If I'm using a screen reader, the alt-text doesn't give me enough and the key components list doesn't describe how things function as a system.

maxwelldb commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-architecture.adoc:
```diff
@@ -0,0 +1,51 @@
+
+toc::[]
+
+The following diagram shows the OVN-Kubernetes architecture.
+
+.OVN-Kubernetes architecture
+image::299_OpenShift_OVN-Kubernetes_arch_0123_1.png[OVN-Kubernetes architecture]
+
+The key components are:
+
+* **Cloud Management System (CMS)** - A platform specific client for OVN that provides a CMS specific plugin for OVN integration. The plugin translates the cloud management system's concept of the logical network configuration, stored in the CMS configuration database in a  CMS-specific  format, into an intermediate representation understood by OVN.
+* **OVN Northbound database (`nbdb`)** - Stores the logical network configuration passed by the CMS plugin.
+* **OVN Southbound database (`sbdb`)** - Stores the physical and logical network configuration state for OpenVswitch (OVS) system on each node, including tables that bind them.
+* **ovn-northd** - This is the intermediary client between `nbdb` and `sbdb`. It translates  the logical network configuration in terms of conventional network concepts, taken from the `nbdb`, into  logical data path flows in the `sbdb` below it. The container name is `northd` and it runs in the `ovnkube-master` pods.
+* **ovn-controller** - This is the OVN agent that interacts with OVS and hypervisors, for any information or update that is needed for `sbdb`. The `ovn-controller` reads logical flows from the `sbdb`, translates them into `OpenFlow` flows and sends them to the node’s OVS daemon. The container name is `ovn-controller` and it runs in the `ovnkube-node` pods.
+
+The OVN northbound database has the logical network configuration passed down to it by the cloud management system (CMS).
+The OVN northbound Database contains the current desired state of the network, presented as a collection of logical ports, logical switches, logical routers, and more.
+The `ovn-northd` (`northd` container) connects to the OVN northbound database and the OVN southbound database.
+It translates the logical network configuration in terms of conventional network concepts, taken from the OVN northbound Database, into logical data path flows in the OVN southbound database.
+
+The OVN southbound database has physical and logical representations of the network and binding tables that link them together. Every node in the cluster is represented in the southbound database, and you can see the ports that are connected to it.
+It also contains all the logic flows, the logic flows are shared with the `ovn-controller` process that runs on each node and the `ovn-controller` turns those into `OpenFlow` rules to program `Open vSwitch`.
```
This information should exist in a concept module. Maybe a reference.

kquinn1204 replied:
done

maxwelldb reviewed (commented):
I reviewed the remaining modules in the PR, as well as an assembly.

I did not do much copy editing--I concentrated on the mod docs and repo guidelines that I think need attention.

kquinn1204 commented on modules/nw-ovn-kubernetes-alerts.adoc:
```diff
@@ -0,0 +1,52 @@
+
+* You have access to the cluster as a developer or as a user with view permissions for the project that you are viewing metrics for.
+
+.Procedure (UI)
```
done

kquinn1204 commented on modules/nw-ovn-kubernetes-alerts.adoc:
```diff
@@ -0,0 +1,52 @@
+
+.Procedure (UI)
+
+. In the *Administrator* perspective, select *Observe* -> *Alerting*. The three main pages in the Alerting UI in this perspective are the *Alerts*, *Silences*, and *Alerting Rules* pages.
```
added intro line

kcarmichael08 commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-tracing-using-ovntrace.adoc:
```diff
@@ -0,0 +1,29 @@
+
+toc::[]
+
+OVN and OVS traffic flows can be simulated in a single utility called `ovnkube-trace`. The `ovnkube-trace` utility runs `ovn-trace`, `ovs-appctl ofproto/trace` and `ovn-detrace` and correlates all that information in one easy to read output.
```
```suggestion
OVN and OVS traffic flows can be simulated in a single utility called `ovnkube-trace`. The `ovnkube-trace` utility runs `ovn-trace`, `ovs-appctl ofproto/trace` and `ovn-detrace` and correlates that information in a single output.
```

kcarmichael08 commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-tracing-using-ovntrace.adoc:
```diff
@@ -0,0 +1,29 @@
+
+OVN and OVS traffic flows can be simulated in a single utility called `ovnkube-trace`. The `ovnkube-trace` utility runs `ovn-trace`, `ovs-appctl ofproto/trace` and `ovn-detrace` and correlates all that information in one easy to read output.
+
+The `ovnkube-trace` binary can be executed from a dedicated container, or since {product-title} 4.7, the binary can be copied to a local host and executed from there.
```
```suggestion
You can execute the `ovnkube-trace` binary from a dedicated container. For releases after {product-title} 4.7, you can also copy the binary to a local host and execute it from that host.
```

kcarmichael08 replied:
suggestion: avoid passive voice

kcarmichael08 commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-tracing-using-ovntrace.adoc:
```diff
@@ -0,0 +1,29 @@
+
+[NOTE]
+====
+The binary in the quay images do not currently work for Dual IP stack or IPv6 only environments, for those environments it is required to build from source.
```
```suggestion
The binaries in the Quay images do not currently work for Dual IP stack or IPv6 only environments. For those environments, you must build from source.
```

kcarmichael08 commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-tracing-using-ovntrace.adoc:
```diff
@@ -0,0 +1,29 @@
+The binary in the quay images do not currently work for Dual IP stack or IPv6 only environments, for those environments it is required to build from source.
+====
+
+include::modules/nw-ovn-kubernetes-install-ovnkube-trace-local.adoc[leveloffset=+1]
```
module reviewed by Max

kcarmichael08 commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-tracing-using-ovntrace.adoc:
```diff
@@ -0,0 +1,29 @@
+
+include::modules/nw-ovn-kubernetes-install-ovnkube-trace-local.adoc[leveloffset=+1]
+
+include::modules/nw-ovn-kubernetes-running-ovnkube-trace.adoc[leveloffset=+1]
```
reviewed by Max

kcarmichael08 commented on networking/ovn_kubernetes_network_provider/ovn-kubernetes-troubleshooting-sources.adoc:
```diff
@@ -0,0 +1,35 @@
+:_content-type: ASSEMBLY
+[id="ovn-kubernetes-sources-of-troubleshooting-information"]
+= OVN-Kubernetes sources of troubleshooting information
```
Would a title like "Troubleshooting OVN-Kubernetes" or something more active work better?

kcarmichael08 reviewed (commented):
I had a few comments/suggestions on these.

kquinn1204 commented on modules/nw-ovn-kubernetes-examine-database-contents-ref.adoc:
```diff
@@ -0,0 +1,73 @@
+
+OVN components may be examined with the `ovn-nbctl` and `ovn-sbctl` utilities.
+
+To run OVN commands, find the `ovnkube-master-xxxx` raft leader pod. See _Listing OVN northbound database contents with ovn-kubernetes_ and _Listing OVN southbound database contents with ovn-kubernetes_ for guidance on finding the raft leader for the northbound and southbound database.
+
+When you have identified the raft leader for the northbound database run the following command to show the northbound database contents:
+
+[source,terminal]
+----
+$ oc exec -n openshift-ovn-kubernetes -it ovnkube-master-gt4ms -c northd -- ovn-nbctl show
+----
+
+When you have identified the raft leader for the southbound database run the following command to show the southbound database contents:
+
+[source,terminal]
+----
+$ oc exec -n openshift-ovn-kubernetes -it ovnkube-master-mk6p6 -c northd -- ovn-sbctl show
+----
```
Reorganized this section I split into two sections

kquinn1204 commented:
Do not merge this yet as I am slowly making the updates

gkopels commented on modules/nw-ovn-kubernetes-alerts-cli.adoc:
```diff
@@ -0,0 +1,54 @@
++
+[source,terminal]
+----
+$ curl -s -k -H "Authorization: Bearer \
```
Error: must specify one of -f and -k
error: unknown command "token prometheus-k8s"
See 'oc create -h' for help and examples
parse error: Invalid numeric literal at line 2, column 10

gkopels commented on modules/nw-ovn-kubernetes-alerts-cli.adoc:
```diff
@@ -0,0 +1,54 @@
++
+[source,terminal]
+----
+$ oc get --server="https://$PROMETHEUS_ROUTE" --raw="/api/v1/rules"  | jq '.data.groups[].rules[] | select(((.name|contains("ovn")) or (.name|contains("OVN")) or (.name|contains("Ovn")) or (.name|contains("North")) or (.name|contains("South"))) and .type=="alerting")'
```
Unable to connect to the server: x509: certificate signed by unknown authority

CFields651 commented on modules/ovn-kubernetes-logical-architecture-con.adoc:
```diff
@@ -0,0 +1,68 @@
+Gateway routers:: Gateway routers sometimes called L3 gateway routers, are typically used between the distributed routers and the physical network. Gateway routers including their logical patch ports are bound to a physical location (not distributed), or chassis. The patch ports on this router are known as l3gateway ports in the ovn-southbound database (`ovn-sbdb`). For example in the following ` ...
++
+[source,terminal]
+----
```
I think it's more practical to give a command that customers can more easily repeat for this like:
[root@openshift-jumpserver-0 debug-scripts]# ./network-tools ovn-db-run-command ovn-nbctl lr-list
Leader pod is ovnkube-master-nbrmw
74ccc2cc-6e38-49a7-bcad-c40eea2345f0 (GR_openshift-master-0)
77ce4ade-ddbc-4fc3-820b-f6b1424cbd4a (GR_openshift-master-1)
846bdb13-55d3-4165-bfb0-b7d290477c9e (GR_openshift-master-2)
d6a6bfc6-6ab4-4b55-9de7-abe03087b09b (GR_openshift-worker-0)
f62c6d6f-8818-49ca-be4c-b8df68987855 (GR_openshift-worker-1)
2842e138-66cb-404f-9031-c6afab146be6 (GR_openshift-worker-2)
51394e7e-f57e-4271-ae5b-896848ac5160 (ovn_cluster_router).   

Of course this does show both types of routers.

kquinn1204 replied:
@CFields651 I do tend to agree with you, but I believe we discourage the installation of third party tool in our documentation. I know they are support concerns around using jq so I expect there may be issues with network-tools though maybe not. I will check with OCP doc team

CFields651 commented on modules/ovn-kubernetes-logical-architecture-con.adoc:
```diff
@@ -0,0 +1,68 @@
+Gateway routers:: Gateway routers sometimes called L3 gateway routers, are typically used between the distributed routers and the physical network. Gateway routers including their logical patch ports are bound to a physical location (not distributed), or chassis. The patch ports on this router are known as l3gateway ports in the ovn-southbound database (`ovn-sbdb`). For example in the following ` ...
++
+[source,terminal]
+----
```
I think it's more practical to give a command that customers can more easily repeat for this like:
[root@openshift-jumpserver-0 debug-scripts]# ./network-tools ovn-db-run-command ovn-nbctl lr-list
Leader pod is ovnkube-master-nbrmw
74ccc2cc-6e38-49a7-bcad-c40eea2345f0 (GR_openshift-master-0)
77ce4ade-ddbc-4fc3-820b-f6b1424cbd4a (GR_openshift-master-1)
846bdb13-55d3-4165-bfb0-b7d290477c9e (GR_openshift-master-2)
d6a6bfc6-6ab4-4b55-9de7-abe03087b09b (GR_openshift-worker-0)
f62c6d6f-8818-49ca-be4c-b8df68987855 (GR_openshift-worker-1)
2842e138-66cb-404f-9031-c6afab146be6 (GR_openshift-worker-2)
51394e7e-f57e-4271-ae5b-896848ac5160 (ovn_cluster_router).   

Of course this does show both types of routers.

CFields651 replied:
I'm using a tool from https://github.com/openshift/network-tools to make finding the master pod easier.  the important part of that command is 'ovn-nbctl lr-list'

kquinn1204 replied:
@CFields651 I do tend to agree with you, but I believe we discourage the installation of third party tool in our documentation. I know they are support concerns around using jq so I expect there may be issues with network-tools though maybe not. I will check with OCP doc team

CFields651 commented on modules/ovn-kubernetes-logical-architecture-con.adoc:
```diff
@@ -0,0 +1,68 @@
+  *     "ext_ip-10-0-206-73.us-east-2.compute.internal" (98366c43-7d8d-4b6d-8d83-0a64a68175a2) [ingress]
+----
+
+Logical switches with patch ports:: Logical switches with patch ports are used to virtualize the network stack. They connect remote logical ports through tunnels.
```
To find the 'patch' ports:
 ./network-tools ovn-db-run-command ovn-sbctl find Port_Binding type=patch

CFields651 commented on modules/ovn-kubernetes-logical-architecture-con.adoc:
```diff
@@ -0,0 +1,68 @@
+
+Logical switches with patch ports:: Logical switches with patch ports are used to virtualize the network stack. They connect remote logical ports through tunnels.
+
+Logical switches with localnet ports:: Logical switches with localnet ports are used to connect OVN to the physical network. They connect remote logical ports by bridging the packets to directly connected physical L2 segments using localnet ports. For example in the following `ovnkube-trace` snippet the join logical switches with localnet ports are prefixed with `ext_`.
```
To find 'localnet' ports:
./network-tools ovn-db-run-command ovn-sbctl find Port_Binding type=localnet

kquinn1204 replied:
@CFields651 I do tend to agree with you, but I believe we discourage the installation of third party tool in our documentation. I know they are support concerns around using jq so I expect there may be issues with network-tools though maybe not. I will check with OCP doc team

CFields651 commented on modules/ovn-kubernetes-logical-architecture-con.adoc:
```diff
@@ -0,0 +1,68 @@
+
+Patch ports:: Patch ports represent connectivity between logical switches and logical routers and between peer logical routers. A single connection has a pair of patch ports at each such point of connectivity, one on each side.
+
+l3gateway ports:: l3gateway ports are the port binding entries in the `ovn-sbdb` for logical patch ports used in the gateway routers. They are called l3gateway ports rather than patch ports just to portray the fact that these ports are bound to a chassis just like the gateway router itself.
```
To find l3gateway ports:
./network-tools ovn-db-run-command ovn-sbctl find Port_Binding type=l3gateway

CFields651 commented on modules/ovn-kubernetes-logical-architecture-con.adoc:
```diff
@@ -0,0 +1,68 @@
+  *     "ext_ip-10-0-149-128.us-east-2.compute.internal" (d5481d3e-fac2-47b7-a218-71497dbc6e33) [ingress]
+----
+
+Patch ports:: Patch ports represent connectivity between logical switches and logical routers and between peer logical routers. A single connection has a pair of patch ports at each such point of connectivity, one on each side.
```
To find the 'patch' ports:
./network-tools ovn-db-run-command ovn-sbctl find Port_Binding type=patch

kquinn1204 replied:
@CFields651 I do tend to agree with you, but I believe we discourage the installation of third party tool in our documentation. I know they are support concerns around using jq so I expect there may be issues with network-tools though maybe not. I will check with OCP doc team

CFields651 commented on modules/ovn-kubernetes-logical-architecture-con.adoc:
```diff
@@ -0,0 +1,68 @@
+:_content-type: CONCEPT
+[id="ovn-kubernetes-logical-architecture-con_{context}"]
+= OVN-Kubernetes logical architecture
```
We chose not to include this diagram: https://docs.google.com/drawings/d/1cdjt_fU3DxYbv1XExW9IKG-G3KgoifdFv1WytSJVHJ8/edit?usp=sharing.  Too bad, I think it really fits well in this section because it has labels on the types of switches, routers and ports.  Could we perhaps get this in the graphics design pipeline and publish without it and then get it in the doc when it's complete?

kquinn1204 replied:
@CFields651 I think we agreed on the first two but I think we can address this in a follow up PR. I will get this into the graphics design pipeline

kquinn1204 commented on modules/nw-ovn-kubernetes-alerts-cli.adoc:
```diff
@@ -0,0 +1,54 @@
++
+[source,terminal]
+----
+$ curl -s -k -H "Authorization: Bearer \
```
@gkopels I ran these directly from the HTML and the copy function pasting into cmd line and they worked as expected.

kquinn1204 commented on modules/nw-ovn-kubernetes-alerts-cli.adoc:
```diff
@@ -0,0 +1,54 @@
++
+[source,terminal]
+----
+$ oc get --server="https://$PROMETHEUS_ROUTE" --raw="/api/v1/rules"  | jq '.data.groups[].rules[] | select(((.name|contains("ovn")) or (.name|contains("OVN")) or (.name|contains("Ovn")) or (.name|contains("North")) or (.name|contains("South"))) and .type=="alerting")'
```
@gkopels I ran these directly from the HTML and the copy function pasting into cmd line and they worked as expected.

kquinn1204 commented on modules/nw-ovn-kubernetes-logs-cli.adoc:
```diff
@@ -0,0 +1,66 @@
++
+[source,terminal]
+----
+$ oc logs ovnkube-master-7h4q7 -n openshift-ovn-kubernetes
```
@gkopels I ran these directly from the HTML and the copy function pasting into cmd line and they worked as expected.

gkopels commented:
lgtm

kquinn1204 commented:
/label peer-review-needed

kquinn1204 commented:
I addressed comments in first peer review I have added two new sections nw-ovn-kubernetes-installing-network-tools.adoc and nw-ovn-kubernetes-running-network-tools.adoc

jldohmann commented on modules/nw-ovn-kubernetes-running-network-tools.adoc:
```diff
@@ -0,0 +1,223 @@
+.Prerequisites
+
+* You installed the OpenShift CLI (`oc`).
+* You are logged in to the cluster with a user with `cluster-admin` privileges.
```
```suggestion
* You are logged in to the cluster as a user with `cluster-admin` privileges.
```

jldohmann commented on modules/nw-ovn-kubernetes-running-network-tools.adoc:
```diff
@@ -0,0 +1,223 @@
+[...]
+----
+
+. List the l3gateway ports by running the following command:
```
is l3gateway the common way to refer to this? Consider the following:
```suggestion
. List the L3 gateway ports by running the following command:
```
or maybe it makes more sense to keep the original, but put it in backticks since it seems to be referring to a command option:
```suggestion
. List the `l3gateway` ports by running the following command:
```

jldohmann reviewed (commented):
Generally lgtm, some minor nits below. This review is just for the new files:

- `nw-ovn-kubernetes-installing-network-tools.adoc`
- `nw-ovn-kubernetes-running-network-tools.adoc`

kquinn1204 commented:
/label merge-review-needed

mburke5678 commented:
Hold on merge. Need updated [graphic here](https://54155--docspreview.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/ovn-kubernetes-architecture-assembly.html).

kquinn1204 commented:
/label merge-review-needed

jeana-redhat commented:
/cherrypick enterprise-4.13

jeana-redhat commented:
/cherrypick enterprise-4.12
